raiden/connection_manager.py

```diff
@@ -126,34 +143,48 @@ def leave(self, wait_for_settle=True, timeout=30):
                 c.partner_address) for c in open_channels]
             for channel in channel_specs:
                 try:
-                    self.api.close(*channel),
-                except RuntimeError:
+                    self.api.close(*channel)
+                except:
                     # if the error wasn't that the channel was already closed: raise
                     if channel[1] in [c.partner_address for c in self.open_channels]:
                         raise
 
             # wait for events to propagate
             gevent.sleep(self.raiden.alarm.wait_time)
 
-            if wait_for_settle:
-                try:
-                    with gevent.timeout.Timeout(timeout):
-                        while any(c.state != CHANNEL_STATE_SETTLED for c in open_channels):
-                            # wait for events to propagate
-                            gevent.sleep(self.raiden.alarm.wait_time)
-
-                except gevent.timeout.Timeout:
-                    log.debug(
-                        'timeout while waiting for settlement',
-                        unsettled=sum(
-                            1 for channel in open_channels if
-                            channel.state != CHANNEL_STATE_SETTLED
-                        ),
-                        settled=sum(
-                            1 for channel in open_channels if
-                            channel.state == CHANNEL_STATE_SETTLED
-                        )
+        if wait_for_settle:
+            not_settled_channels = [
+                channel for channel in
+                self.api.get_channel_list(token_address=self.token_address)
+                if channel.state != CHANNEL_STATE_SETTLED
+            ]
+            try:
+                with gevent.timeout.Timeout(timeout):
+                    while any(c.state != CHANNEL_STATE_SETTLED for c in not_settled_channels):
+                        # wait for events to propagate
+                        gevent.sleep(self.raiden.alarm.wait_time)
+
+            except gevent.timeout.Timeout:
+                async_result.set(False)
+                log.debug(
+                    'timeout while waiting for settlement',
+                    unsettled=sum(
```

Comment:
nitpicking: `unsettled` -> `not_settled`. Unsettled has a different meaning than what you would like to convey here.